### ISSUE TITLE
[feature] resizable table

### ISSUE THREAD

**Issue by ghost** [drawdb-io/drawdb#188]
Currently the table component size is fixed, so when the table name is long it'll simply be replaced with 3-dots. It would be nice to resize or auto-resize to accommodate for the full length of the table and column name.

**Comment by 1ilit** [comment]
You can set the table width here
![image](https://github.com/user-attachments/assets/5161c3a7-79b9-4d4b-90ec-9dab345cd0ae)

**Comment by ruizhii** [comment]
I see, thanks for pointing that out @1ilit , I didn't know that. This would solve my problem but ultimately I still think manual or auto-resizable card can be useful, is there a plan to have it or a reason to not? I wouldn't mind to help with building it if this feature can be added to the product

**Comment by 1ilit** [comment]
If the tables were individually resized and were all different sizes I don't think the diagrams would look as good

**Comment by ruizhii** [comment]
You have a point, but I don't think having too much whitespace on smaller table look good either.

![image](https://github.com/user-attachments/assets/d13ea2d2-7376-4fd5-af34-2c33a608a9ca)

**Comment by sahitya-chandra** [comment]
Hi, can i take up this issue ??
If Yes... then can you just elaborate specifically what do you mean by "resizable table" because drawdb already have table width feature.

**Comment by 1ilit** [comment]
@sahitya-chandra I'm still not convinced about having this implemented

**Comment by sudhirVoid** [comment]
> @sahitya-chandra I'm still not convinced about having this implemented

What I think is in some table the column name can be long so it is shown as ... In that case I have to increase the width of the whole diagram where the proper management of space will not be there. So, I think the existing feature is very good to set the default table size but at table level too we have to give functionality to drag and expand width. I think.

**Comment by sudhirVoid** [comment]
> If the tables were individually resized and were all different sizes I don't think the diagrams would look as good

What I think is in some table the column name can be long so it is shown as ... In that case I have to increase the width of the whole diagram where the proper management of space will not be there. So, I think the existing feature is very good to set the default table size but at table level too we have to give functionality to drag and expand width. I think.

**Comment by hkusulja** [comment]
I suggest that each table has its width settings / property.
Auto
Or fixed
or inherited from general settings

**Comment by sudhirVoid** [comment]
> I suggest that each table has its width settings / property. Auto Or fixed or inherited from general settings

Yes this is what I also want to achieve here but in the form of drag and expand from the corners . This will reinitialize the width of that specific table again . I just see that having 2 - 3 clicks just to set width can be irritating if tables are many. So for the ease drag and expand can be the right choice. 
what you think?

**Comment by JamalArshad147** [comment]
I believe it should be the user's choice. If we have enabled them to change the table size in settings then we should also offer a convenient way. If someone prefers to maintain a specific size, that is a matter of personal preference. If someone wants to resize it then he is allowed to do it. This way we will give them the choice or customization option to resize it.

**Comment by wowkin2** [comment]
Would really appreciate to have this feature.
Because some of my tables have long names and also enums as types, and others are 2-3 times smaller, but I can't use that space.

**Comment by EnriekNP** [comment]
Hi maintainers 👋
I’m interested in working on this feature (resizable tables) since it’s something I personally need as well.
Would it be okay if I take this issue and submit a PR?
Happy to discuss the approach first if needed.
